### ISSUE TITLE
contextual menus upgrade

### ISSUE THREAD

**Issue by bradleyrichter** [brave/browser-laptop#1253]
Brave users will expect familiar contextual menus that include the commonly used items found in most browsers, plus convienent access to new Brave features.

Please implement as shown in the spec images below: 

![image](https://cloud.githubusercontent.com/assets/13509546/14269062/e450c646-fa95-11e5-9939-bcc2cea16576.png)

![image](https://cloud.githubusercontent.com/assets/13509546/14269066/ea5951e8-fa95-11e5-822e-3a13516088c8.png)

![image](https://cloud.githubusercontent.com/assets/13509546/14269071/fc2a979c-fa95-11e5-97e4-0bc8d0cbb4bd.png)

![image](https://cloud.githubusercontent.com/assets/13509546/14269075/04967df6-fa96-11e5-9ecc-e5bab5386698.png)

**Comment by TheDigitalOrchard** [comment]
You've done quite a bit of work preparing these mockups. However, I personally disagree with some of the proposed menu labels. "Open Link in New Tab", for example, You're not opening a "link", you're opening the page that the link points to... a "page". Therefore, "Open in New Tab" is better wording, in my opinion.

I do agree with many of the other changes, particularly small stuff like "Proper Title Case".

**Comment by bradleyrichter** [comment]
@TheDigitalOrchard thanks for the feedback. I'm all for improving things, and correcting lingo. But in this case, is it better to follow than lead? Safari, FX and Chrome all use "Open Link in..."

![image](https://cloud.githubusercontent.com/assets/13509546/14328797/1ee67960-fbed-11e5-8031-18a51983189b.png)

Maybe "Open Linked Page in..." would be more correct but takes more space.

**Comment by weems** [comment]
I think New Tab is general enough, people will know that pages are links. Short and sweet.

**Comment by bsclifton** [comment]
Checking this one out... :smiley: might have some questions

**Comment by diracdeltas** [comment]
@bsclifton how's it going on this? i can take it but don't want to duplicate work.

**Comment by bsclifton** [comment]
@diracdeltas things are going great :smile: I hope to wrap this up tonite.  I'll be on Slack

**Comment by bsclifton** [comment]
@bradleyrichter in the first spec image (NEW MENU: on link), what does "Copy" do (versus copy link)?

**Comment by bradleyrichter** [comment]
I think it copies the selected rich text to clipboard as if it were not linked. The existing menu has it as well.

**Comment by bsclifton** [comment]
Most changes done... Could definitely use help with the remaining items
https://github.com/bsclifton/browser-laptop/commit/b05a82ccf0183af1ba481afb9ec4d324737d2d0d

**Comment by bsclifton** [comment]
Just noticed this... (not related to my changes). The current HEAD for master has a problem with the context menus (inside the hamburger menu):
![hamburger](https://cloud.githubusercontent.com/assets/4733304/14910970/c92d6418-0da6-11e6-9749-1b8cb2ef4b03.png)

**Comment by bbondy** [comment]
> in the first spec image (NEW MENU: on link), what does "Copy" do (versus copy link)?

Copy should copy the text, on a link it will select and copy it.
(Works same in Chrome)


Re the menu problem:
I don't see that, if you could do a pull request that fixes it independent of this, that'd be awesome though. 0.9.3 is on Wednesday.

**Comment by bsclifton** [comment]
> Re the menu problem:
I don't see that, if you could do a pull request that fixes it independent of this, that'd be awesome though. 0.9.3 is on Wednesday.

Perfect- I will look at that next

**Comment by bsclifton** [comment]
Help needed with the commented out items in https://github.com/brave/browser-laptop/pull/1522 cc: @diracdeltas 

Also need help adding the bravery menu at
https://github.com/brave/browser-laptop/blob/f4728589a41844fc368014a60cbfe2d4f86b0a08/js/contextMenus.js#L697

**Comment by bbondy** [comment]
> Tab context menu, move tab to new window

This one is going to be a bit harder for now until @bridiver gets to moving webviews across windows.  We also want it for shared pinned tabs by the way.

> Search selection w/ default search engine

`windowActions.newFrame` w/  appState's `searchDetail.searchURL`  replacing {searchTerms} with the selected text.  I think that might be `window.getSelection().toString()`

> Search highlighted link w/ default search engine

I think the same but use the link text


> adding the bravery menu

This should be similar I think to what the hamburger menu does w/ Bravery.

> Tab context menu, adding "Mute other tabs"

See `muteAll` which you probably wan to pull up higher and then add another parameter for the frame to skip over in the `forEach` loop.

**Comment by diracdeltas** [comment]
>  > adding the bravery menu
> This should be similar I think to what the hamburger menu does w/ Bravery.

I thought the bravery context menu was bravery settings for the current page only, not globally. If so it might be best to leave it out until per-site Bravery settings is implemented.

**Comment by bbondy** [comment]
that should be hooked up by 5/20.

**Comment by bradleyrichter** [comment]
it should wait. Also, it will need to invoke a reload on the current page after the setting change is selected. (unless anyone knows of an architectual reason not to reload with the changed settings?)

**Comment by bbondy** [comment]
Not against it fully but just a thought that someone might have a form they filled out that they can't submit, so try turning off adblock, and then they get a reload losing their form data.   Perhaps prompting for that case.  But that's out of scope of this ticket, so no need to discuss it here.

**Comment by bsclifton** [comment]
Added "Mute other Tabs" w/ https://github.com/brave/browser-laptop/pull/1535

Sounds like we'll have to wait on the bravery menu; I'll check out the search items next

**Comment by bsclifton** [comment]
Added search to context menus w/ https://github.com/brave/browser-laptop/pull/1672

**Comment by bradleyrichter** [comment]
@bbondy @bsclifton 

Not sure how it happened but I unintentionally omitted 2 menu items in the new menu for Tab-Context. Here is a new image:

![image](https://cloud.githubusercontent.com/assets/13509546/15294684/21534a92-1b42-11e6-9d60-5e86751d00a4.png)







I also changed the grouping on the page background context menu as requested by Brian:

![image](https://cloud.githubusercontent.com/assets/13509546/15294785/7d985130-1b42-11e6-822e-b748b2cd9efa.png)

**Comment by bbondy** [comment]
Look up selection is being tracked here:
https://github.com/brave/browser-laptop/issues/1627

**Comment by bbondy** [comment]
The items mentioned here: https://github.com/brave/browser-laptop/issues/1253#issuecomment-219460703 are fixed.

**Comment by bsclifton** [comment]
I submitted https://github.com/brave/browser-laptop/pull/2329 which fixes the casing issues with context menus. They should all be in a consistent Title Case. This PR also matches Chrome for the order of context menu items for situations that were not mocked up

I'd like to propose closing this issue if the above PR is accepted :smile: With the long history here, maybe we can open issues for the remaining items:
1) Context menu when right clicking tabs => should show "Move to New Window"
2) Context menu for page should show the Bravery menu for the active window

Anything else missing? The lion's share (LOL) of this issue has been handled

**Comment by bsclifton** [comment]
Will look at closing out the copy image to clipboard with https://github.com/brave/browser-laptop/issues/1174#issuecomment-229832864

@bradleyrichter would it make sense to close this issue out and make two new ones for the above cases? If so, LMK and I'll take care of that

**Comment by bsclifton** [comment]
Closing this issue in favor of tasks tracking the remaining work:
https://github.com/brave/browser-laptop/issues/3129
https://github.com/brave/browser-laptop/issues/3130
